apps/ledger-live-mobile/src/families/ethereum/EditTransaction/MethodSelection.tsx

```diff
@@ -0,0 +1,277 @@
+  const bridge = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = useCallback(
+    (option: "cancel" | "speedup") => {
+      switch (option) {
+        case "cancel":
+          {
+            const mainAccount = getMainAccount(account, parentAccount);
+            const updatedTransaction: Partial<Transaction> = {
+              amount: new BigNumber(0),
+              data: Buffer.from(""),
+              nonce: operation.transactionSequenceNumber,
+              allowZeroAmount: true,
+              mode: "send",
+              recipient: mainAccount.freshAddress,
+            };
+
+            if (EIP1559ShouldBeUsed(mainAccount.currency)) {
+              if (transactionToEdit.maxPriorityFeePerGas) {
+                updatedTransaction.maxPriorityFeePerGas =
+                  transactionToEdit.maxPriorityFeePerGas?.times(
+                    1 +
+                      getEnv(
+                        "EDIT_TX_EIP1559_MAXPRIORITYFEE_GAP_SPEEDUP_FACTOR",
+                      ),
+                  );
+              }
+
+              if (transactionToEdit.maxFeePerGas) {
+                updatedTransaction.maxFeePerGas =
+                  transactionToEdit.maxFeePerGas?.times(
+                    1 + getEnv("EDIT_TX_EIP1559_MAXFEE_GAP_CANCEL_FACTOR"),
+                  );
+              }
+            } else if (transactionToEdit.gasPrice) {
+              updatedTransaction.gasPrice = transactionToEdit.gasPrice.times(
+                1 + getEnv("EDIT_TX_NON_EIP1559_GASPRICE_GAP_CANCEL_FACTOR"),
+              );
+            }
+
+            setTransaction(
+              bridge.updateTransaction(transaction, updatedTransaction),
+            );
+
+            setSelectedMethod("cancel");
+          }
+
+          break;
+
+        case "speedup":
+          {
+            const updatedTransaction: Partial<Transaction> = {
+              amount: transactionToEdit.amount,
+              data: transactionToEdit.data,
+              recipient: transactionToEdit.recipient,
+              mode: transactionToEdit.mode,
+              nonce: operation.transactionSequenceNumber,
+              // set "fast" as default option for speedup flow
+              feesStrategy: "fast",
+              networkInfo: null,
+              gasPrice: null,
+              maxFeePerGas: null,
+              maxPriorityFeePerGas: null,
+            };
+
+            setTransaction(
+              bridge.updateTransaction(transaction, updatedTransaction),
+            );
+
+            setSelectedMethod("speedup");
+          }
+
+          break;
+        default:
+          break;
+      }
+    },
+    [
+      account,
+      parentAccount,
+      transaction,
+      transactionToEdit,
+      bridge,
+      operation.transactionSequenceNumber,
+      setTransaction,
+      mainAccount.currency,
+    ],
+  );
+
+  apiForCurrency(mainAccount.currency)
+    .getTransactionByHash(operation.hash)
+    .then(tx => {
```

Comment:
As @lambertkevin mentioned on my PR too. I think we should check this on "every" page in the edit eth tx flow.